pkg/model/dev.go

```diff
@@ -156,15 +169,50 @@ func (b *BuildInfo) GetDockerfilePath() string {
 	return joinPath
 }
 
-// AddArgs add a set of args to the build information
-func (b *BuildInfo) AddArgs(args map[string]string) {
-	for k, v := range args {
-		b.Args = append(b.Args, EnvVar{
+// ExpandBuildArgs add a set of args to the build information
+func (b *BuildInfo) ExpandBuildArgs(previousImageArgs map[string]string) error {
```

Comment:
This function is doing more than expanding the build args, it is also (or potentially) including new build args, so maybe we can call it `AddAndExpandBuildArgs`

Reply:
Should we split it in two and use two different functions instead? I don't really like the name `AddAndExpandBuildArgs`.
We can have `AddBuildArgs` and `ExpandBuildArgs` which is more descriptive but I'm afraid that every time we do one we must do the other one so that's why I included them in a single function 🤔

Reply:
If they have to be called together, I would create a single function containing both of them, but given our code base, if I see a function called `ExpandBuildArgs` I wouldn't expect to have more arguments, I would just expect that the existent ones would be expanded.

Other option could be to just call it `AddBuildArgs`. The fact that the build args has to be expanded is internal to `BuildInfo` struct, so if everytime I include some new argument, it will be expanded, I think it is fair to call it just `AddBuildArgs`. In the function comment you can mention that the arguments will be expanded